eth/gaspricemonitor.go

```diff
@@ -59,9 +59,25 @@ func (gpm *GasPriceMonitor) GasPrice() *big.Int {
 	gpm.gasPriceMu.RLock()
 	defer gpm.gasPriceMu.RUnlock()
 
+	if gpm.gasPrice.Cmp(gpm.minGasPrice) < 0 {
```

Comment:
We were already dropping polled gas prices that were below the min gas price, but this ensures that the lowest value returned by GasPrice() is min gas price.